feat: add translation feature documentation (#910)
novuhq/docs: content/docs/platform/workflow/translations.mdx

  },
  "email": {
    "subject": "Get Started",
    "body": "Thank you for joining us, {{subscriber.fullName}}"
  }
}
```

You can export and re-upload these JSON files when translation updates are needed or use the JSON editor on the translation page to update the translations.

![Translation JSON format](/images/platform/workflow/translations/translations_locale_content_edit.gif)

### Exporting and importing translations

#### Export

- Navigate to the Translations section.
- Click **Export en_US.json** to download the latest default file.
- Use this as a base for other translations.

#### Import

- Click **Import locale(s)**.
- Upload JSON files (e.g., fr_FR.json, es_ES.json).
- Novu detects the locale automatically.

💡 Translation changes in default locale (e.g., en_US) will mark other locales as *Outdated, needs update*

## Using translation keys in step content editor

In Email or In-App editors, use `{{t.key}}` to insert a translatable string. If the key already exists for the locale, it appears in the suggestions list.

Clicking **Add to Translations** automatically adds a new key to default locale.

Example:

```html

> {{t.subject}} → "Let's get you started"

Reviewer: We also support nested keys like `{{t.title.button.submit}}`, maybe worth mentioning

---
title: 'Translations'
description: 'Learn how to translate your workflow step content into multiple languages'
icon: 'Languages'
---

## Workflow translations

> Novu supports multi-language translations for workflows, allowing your notifications to dynamically adapt to your users' preferred language. This documentation covers everything you need to know to enable, configure, and manage translations for your workflows.

Reviewer 2: I am trying to help us adapt to Novu's specific wording to fit into our language. So instead of saying users, we called them subscribers

> Novu supports multi-language translations for workflows, allowing your notifications to dynamically adapt to your users' preferred language. This documentation covers everything you need to know to enable, configure, and manage translations for your workflows.

Reviewer 2: ```suggestion
Novu supports multi-language translations for workflows, allowing notifications to dynamically adapt to each of your subscriber's preferred language. You can enable, configure, and manage translations across your notification workflows from the Novu dashboard.
```

<Callout type="info">

>   The translations feature is currently in `beta` and is available for `Team` and `Enterprise` plans.

Reviewer 2: We can add a link to the plans page here.

## Using translation keys in step content editor

In Email or In-App editors, use `{{t.key}}` to insert a translatable string. If the key already exists for the locale, it appears in the suggestions list.

Clicking **Add to Translations** automatically adds a new key to default locale.

Example:

```html
{{t.subject}} → "Let's get you started"

{{t.product.details.description}}
```

![Using translation keys in step content editor](/images/platform/workflow/translations/translations_variable_use_in_step_editor.gif)

## Testing translations

Use the Preview Context to:

- Simulate subscriber locale such as `fr_FR` or `ru_RU`
- Preview step content for the selected language.
- Update or correct missing translations on the fly.

## Keeping translations up to date

Any edit to the default locale will invalidate others. You’ll see a warning: `Outdated, needs update`. Re-export the default, update translated files, and re-upload.

## Disabling Translations

A translation group can be disabled by clicking on the three dots and selecting **Disable & delete translations**. This action is irreversible and removes all associated translation data.

> ## Translation in Layouts

Reviewer 2: ```suggestion
## Translation in email layouts
```
